plugins/woocommerce/src/Blocks/Templates/AbstractTemplate.php

```diff
@@ -0,0 +1,45 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+/**
+ * AbstractTemplate class.
+ *
+ * Shared logic for templates.
+ *
+ * @internal
+ */
+abstract class AbstractTemplate {
+
+	/**
+	 * The slug of the template.
+	 *
+	 * @var string
+	 */
+	const SLUG = '';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->init();
+	}
+
+	/**
+	 * Initialization method.
+	 */
+	abstract public function init();
+
+	/**
+	 * Should return the title of the template.
+	 *
+	 * @return string
+	 */
+	abstract public static function get_template_title();
+
+	/**
+	 * Should return the description of the template.
+	 *
+	 * @return string
+	 */
+	abstract public static function get_template_description();
```

Comment:
Another question about static methods here, from the usage of these methods, I don't see why they need to be static.

Reply:
In this case, the reason is that `AbstractPageTemplate` currently has a static `get_template_title()` method, so I kept it static too in the new class:

https://github.com/woocommerce/woocommerce/blob/8b4bacc2bfbb66950264e197955e68f4b532a07b/plugins/woocommerce/src/Blocks/Templates/AbstractPageTemplate.php#L55

But as I mentioned in the other comment, I don't have a strong opinion and happy to change this. But I'm not sure about the benefits of making these non-static.